pkg/api/types.go

```diff
@@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
 	// Items is a list of ResourceQuota objects
 	Items []ResourceQuota `json:"items"`
 }
+
+// Secret holds secret data of a certain type
+type Secret struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	Data map[string][]byte `json:"data,omitempty"`
+	Type SecretType        `json:"type,omitempty"`
+}
+
+type SecretType string
+
+const (
+	SecretTypeOpaque SecretType = "opaque" // Default; arbitrary user-defined data
```

Comment:
@pmorie /cc me on the PR if it modifies the API that is checked in.  I am working on something that uses this already so I would need to update.

Sent from my iPhone

> On Feb 21, 2015, at 12:28 PM, Paul Morie <notifications@github.com> wrote:
> 
> In pkg/api/types.go:
> 
> > @@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
> >  	// Items is a list of ResourceQuota objects
> >  	Items []ResourceQuota `json:"items"`
> >  }
> > +
> > +// Secret holds secret data of a certain type
> > +type Secret struct {
> > +	TypeMeta   `json:",inline"`
> > +	ObjectMeta `json:"metadata,omitempty"`
> > +
> > +	Data map[string][]byte `json:"data,omitempty"`
> > +	Type SecretType        `json:"type,omitempty"`
> > +}
> > +
> > +type SecretType string
> > +
> > +const (
> > +	SecretTypeOpaque SecretType = "opaque" // Default; arbitrary user-defined data
> @bgrant0607 @thockin will collect these i to a PR soon (might not be until Monday).
> …
> —
> Reply to this email directly or view it on GitHub.
>